libs/remote_storage/src/local_fs.rs

```diff
@@ -562,31 +515,27 @@ impl RemoteStorage for LocalFs {
             })
             .map_err(DownloadError::Other)?;
 
-        let len = source
-            .metadata()
-            .await
-            .context("query file length")
-            .map_err(DownloadError::Other)?
-            .len();
+        let mut take = file_metadata.len();
+        if let Some((start, end)) = opts.byte_range() {
+            file.seek(io::SeekFrom::Start(start))
```

Comment:
the seek is not necessary if `start` is 0, no?

Reply:
True. I figured this was only used in tests, so we don't particularly care about the overhead, but I've made it conditional.